src/matchers.ts

```diff
@@ -16,46 +16,73 @@
-    p.hasTagName('link'), p.hasAttrValue('rel', 'import'),
-    p.hasAttrValue('type', 'css'));
-
-export const styleMatcher: Matcher = p.AND(
-    p.hasTagName('style'),
-    p.OR(p.NOT(p.hasAttr('type')), p.hasAttrValue('type', 'text/css')));
-
-export const targetMatcher: Matcher = p.AND(
-    p.OR(p.hasTagName('a'), p.hasTagName('form')), p.NOT(p.hasAttr('target')));
-
-export const head: Matcher = p.hasTagName('head');
-export const body: Matcher = p.hasTagName('body');
-export const base: Matcher = p.hasTagName('base');
-export const domModule: Matcher = p.AND(
-    p.hasTagName('dom-module'), p.hasAttr('id'), p.NOT(p.hasAttr('assetpath')));
-export const meta: Matcher = p.AND(p.hasTagName('meta'), p.hasAttr('charset'));
-export const polymerElement: Matcher = p.hasTagName('polymer-element');
-export const externalJavascript: Matcher = p.AND(p.hasAttr('src'), jsMatcher);
+export const polymerExternalStyle: Matcher = predicates.AND(
+    predicates.hasTagName('link'), predicates.hasAttrValue('rel', 'import'),
+    predicates.hasAttrValue('type', 'css'));
+
+export const styleMatcher: Matcher = predicates.AND(
+    predicates.hasTagName('style'),
+    predicates.OR(
+        predicates.NOT(predicates.hasAttr('type')),
+        predicates.hasAttrValue('type', 'text/css')));
+
+export const targetMatcher: Matcher = predicates.AND(
+    predicates.OR(predicates.hasTagName('a'), predicates.hasTagName('form')),
+    predicates.NOT(predicates.hasAttr('target')));
+
+export const head: Matcher = predicates.hasTagName('head');
+export const body: Matcher = predicates.hasTagName('body');
+export const base: Matcher = predicates.hasTagName('base');
+export const domModule: Matcher = predicates.AND(
+    predicates.hasTagName('dom-module'), predicates.hasAttr('id'),
+    predicates.NOT(predicates.hasAttr('assetpath')));
+export const meta: Matcher = predicates.AND(
+    predicates.hasTagName('meta'), predicates.hasAttr('charset'));
+export const polymerElement: Matcher = predicates.hasTagName('polymer-element');
+export const externalJavascript: Matcher =
+    predicates.AND(predicates.hasAttr('src'), jsMatcher);
 export const inlineJavascript: Matcher =
-    p.AND(p.NOT(p.hasAttr('src')), jsMatcher);
-export const htmlImport: Matcher = p.AND(
-    p.hasTagName('link'), p.hasAttrValue('rel', 'import'), p.hasAttr('href'));
+    predicates.AND(predicates.NOT(predicates.hasAttr('src')), jsMatcher);
+export const htmlImport: Matcher = predicates.AND(
+    predicates.hasTagName('link'), predicates.hasAttrValue('rel', 'import'),
+    predicates.hasAttr('href'));
+
+export const hiddenDiv = predicates.AND(
+    predicates.hasTagName('div'), predicates.hasAttr('hidden'),
+    predicates.hasAttr('by-vulcanize'));
+
+export const inHiddenDiv = predicates.parentMatches(hiddenDiv);
+const nextToHiddenDiv = (offset) => {
+  return (node) => {
+    const siblings = node.parentNode.childNodes;
+    const hiddenDivIndex = siblings.indexOf(node) + offset;
```

Comment:
This matcher needs to deal with a number of edge cases. Whitespace-only text nodes should be ignored, text nodes with meaningful space should be preserved. Comments should be ignored if --strip-comments is set. License comments should be deduplicated and moved to the start of the document.